packages/medusa/src/models/payment-collection.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  BeforeInsert,
+  Column,
+  Index,
+  JoinColumn,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+} from "typeorm"
+
+import { SoftDeletableEntity } from "../interfaces/models/soft-deletable-entity"
+import { DbAwareColumn } from "../utils/db-aware-column"
+import { generateEntityId } from "../utils"
+import { Currency, Payment, PaymentSession, Region } from "."
+
+import OrderEditingFeatureFlag from "../loaders/feature-flags/order-editing"
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+
+export enum PaymentCollectionStatus {
+  NOT_PAID = "not_paid",
+  AWAITING = "awaiting",
+  AUTHORIZED = "authorized",
+  PARTIALLY_AUTHORIZED = "partially_authorized",
+  CAPTURED = "captured",
+  PARTIALLY_CAPTURED = "partially_captured",
+  REFUNDED = "refunded",
+  PARTIALLY_REFUNDED = "partially_refunded",
+  CANCELED = "canceled",
+  REQUIRES_ACTION = "requires_action",
+}
+
+export enum PaymentCollectionType {
+  ORDER_EDIT = "order_edit",
+  SINGLE_PAYMENT = "single_payment",
```

Comment:
suggestion(non-blocking): Omit this for now as it is of no use
```suggestion
  SINGLE_PAYMENT = "single_payment",
```